### ISSUE TITLE
Address param-nullchecking formatting issues

### ISSUE THREAD

**Issue by RikkiGibson** [dotnet/roslyn#58950]
We have such a test for the formatter itself. It wouldn't hurt to have it for the formatting analyzer too.

https://github.com/dotnet/roslyn/blob/259a9e27ab8c6a1d8d3a3445b079fd5e3a13ec0a/src/Workspaces/CSharpTest/Formatting/FormattingTests.cs#L5779-L5799

I did find that we don't do the right thing in the following scenario:

```cs
static void M1(string x
    !!)
{
}
```

I expected `!!` to always be moved to be right up against the parameter name. It seems like the leading trivia owned by `!!` should be moved to the next token, for consistency with the following (which doesn't prompt any formatting errors)

```cs
static void M1(string x
    )
{
}
```

All that said, I'd prefer to back-burner this work and possibly make it "help wanted". The important thing I wanted to confirm with this PR is that "correctly formatted" code using the feature doesn't produce formatting warnings.

_Originally posted by @RikkiGibson in https://github.com/dotnet/roslyn/issues/58936#issuecomment-1016816761_

**Comment by jcouv** [comment]
Closing as the feature is no longer applicable.
